packages/techdocs-common/src/stages/publish/googleStorage.ts

```diff
@@ -0,0 +1,222 @@
+          );
+          resolve({});
+        });
+    });
+  }
+
+  fetchTechDocsMetadata(entityName: EntityName): Promise<string> {
+    return new Promise((resolve, reject) => {
+      const entityRootDir = `${entityName.namespace}/${entityName.kind}/${entityName.name}`;
+
+      const fileStreamChunks: Array<any> = [];
+      this.storageClient
+        .bucket(this.bucketName)
+        .file(`${entityRootDir}/techdocs_metadata.json`)
+        .createReadStream()
+        .on('error', err => {
+          this.logger.error(err.message);
+          reject(err.message);
+        })
+        .on('data', chunk => {
+          fileStreamChunks.push(chunk);
+        })
+        .on('end', () => {
+          const techdocsMetadataJson = Buffer.concat(
+            fileStreamChunks,
+          ).toString();
+          resolve(techdocsMetadataJson);
+        });
+    });
+  }
+
+  /**
+   * Express route middleware to serve static files on a route in techdocs-backend.
+   */
+  docsRouter(): express.Handler {
+    return (req, res) => {
+      // Trim the leading forward slash
+      // filePath example - /default/Component/documented-component/index.html
+      const filePath = req.path.replace(/^\//, '');
+
+      // Files with different extensions (CSS, HTML) need to be served with different headers
+      const fileExtension = filePath.split('.')[filePath.split('.').length - 1];
+      const responseHeaders = getHeadersForFileExtension(
+        fileExtension as supportedFileType,
+      );
+
+      const fileStreamChunks: Array<any> = [];
+      this.storageClient
+        .bucket(this.bucketName)
+        .file(filePath)
+        .createReadStream()
+        .on('error', err => {
+          this.logger.warn(err.message);
+          res.status(404).send(err.message);
+        })
+        .on('data', chunk => {
+          fileStreamChunks.push(chunk);
+        })
+        .on('end', () => {
+          const fileContent = Buffer.concat(fileStreamChunks).toString();
+          // Inject response headers
+          for (const [headerKey, headerValue] of Object.entries(
+            responseHeaders,
+          )) {
+            res.setHeader(headerKey, headerValue);
+          }
+
+          res.send(fileContent);
+        });
+    };
+  }
+
+  /**
+   * A helper function which checks if index.html of an Entity's docs site is available. This
+   * can be used to verify if there are any pre-generated docs available to serve.
+   */
+  async hasDocsBeenGenerated(entity: Entity): Promise<boolean> {
+    return new Promise(resolve => {
+      const entityRootDir = `${entity.metadata.namespace}/${entity.kind}/${entity.metadata.name}`;
+      this.storageClient
+        .bucket(this.bucketName)
+        .file(`${entityRootDir}/index.html`)
+        .createReadStream()
+        .on('error', () => {
+          resolve(false);
```

Comment:
should probably check the error, right? to distinguish problems from just missing

Reply:
I looked into the documentation for this library and did not find any example where I can see all possible errors. https://googleapis.dev/nodejs/storage/latest/File.html

Resolving to `false` seems fine to me here, since one way or another, this means that TechDocs will not be able to read files from the storage.